File: CoreApiApp/WebApp/Data/Entities/Schedule.cs
namespace CoreApiApp.Data.Entities;

public class Schedule
{
    public int Id { get; set; }
    public Guid Guid { get; set; }
    public Staff Staff { get; set; }
    public int StaffId { get; set; }
    public Shift Shift { get; set; }
    public int ShiftId { get; set; }
    public Department Department { get; set; }
    public int DepartmentId { get; set; }
    public string ScheduledDate { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }
    
}

File: CoreApiAppPython/models/schedule.py
from sqlalchemy import Column, Integer, String, DateTime, ForeignKey, Date
from sqlalchemy.orm import relationship
from sqlalchemy.sql import func
from models.base import Base
import uuid

class Schedule(Base):
    __tablename__ = "schedule"

    id = Column(Integer, primary_key=True, index=True)
    guid = Column(String, default=lambda: str(uuid.uuid4()), unique=True, index=True)
    staff_id = Column(Integer, ForeignKey("staff.id"))
    shift_id = Column(Integer, ForeignKey("shift.id"))
    department_id = Column(Integer, ForeignKey("department.id"))
    department = relationship("Department")
    schedule_date = Column(Date)
    created_at = Column(DateTime(timezone=True), server_default=func.now())
    updated_at = Column(DateTime(timezone=True), onupdate=func.now())

    staff = relationship("Staff", back_populates="schedules")
    shift = relationship("Shift", back_populates="schedules")